lib/mutant/ast/meta.rb

```diff
@@ -81,7 +44,8 @@ def assignment?
         # @api private
         #
         def attribute_assignment?
-          arguments.one? && attribute_assignment_selector?
+          !Types::METHOD_OPERATORS.include?(selector) &&
+          selector.to_s.end_with?(ATTRIBUTE_ASSIGNMENT_SELECTOR_SUFFIX)
```

Comment:
You missed the `_SUFFIX` part in this constant name. The constant evaluates to `=` and the prefix is chosen by the ruby user. I can define any prefix like: `def foo=(value); end` etc.